oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/TokenRevocation.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.auth.oauth2;
+
+import static com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants.ACCESS_TOKEN;
+import static com.linecorp.armeria.internal.common.auth.oauth2.OAuth2Constants.REFRESH_TOKEN;
+
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.internal.common.auth.oauth2.TokenRevocationRequest;
+
+/**
+ * Implements Token Revocation request/response flow,
+ * as per <a href="https://tools.ietf.org/html/rfc7009">[RFC7009]</a>.
+ */
+@UnstableApi
+public final class TokenRevocation {
```

Comment:
- This class doesn't seem to be used anywhere. Is this something you're gonna use in your follow-up PR?
- Is this class meant to be used by a user? If not, we could move it to `internal.common.auth.oauth2` package.

Reply:
Yes, this class is meant to be used directly by the user, either on the server or on the client side. Token revocation capability is an important part of OAuth 2.

Reply:
Excellent. Sounds good to me.